From 16cb53f68586c65d024de78578653e4ec3fa56ea Mon Sep 17 00:00:00 2001
From: Christian Seiler <christian@iwakd.de>
Date: Fri, 4 Mar 2016 18:03:54 +0100
Subject: [PATCH 3/3] adt-virt-qemu: make i386 arch work by default.

os.uname returns e.g. i686 instead of i386, but the qemu command is
called qemu-system-i386. Handle that and add support for generic
mappings for future similar cases.

Reproducer: setarch i386 adt-run ... --- adt-virt-qemu image
---
 debian/changelog           |  1 +
 virt-subproc/adt-virt-qemu | 15 +++++++++++++--
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/debian/changelog b/debian/changelog
index 28c00c0..3a6d8c1 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -23,6 +23,7 @@ autopkgtest (3.19.4) UNRELEASED; urgency=medium
   * Support nested KVM by default by emulating a CPU with VMX/SVM support on
     x86_64.
   * setup-testbed: reduce grub timeout on Debian systems.
+  * adt-virt-qemu: make i386 arch work by default.
 
  -- Martin Pitt <mpitt@debian.org>  Tue, 23 Feb 2016 18:21:51 +0100
 
diff --git a/virt-subproc/adt-virt-qemu b/virt-subproc/adt-virt-qemu
index d53c94e..758f9b1 100755
--- a/virt-subproc/adt-virt-qemu
+++ b/virt-subproc/adt-virt-qemu
@@ -34,6 +34,7 @@ import time
 import socket
 import errno
 import fcntl
+import re
 
 try:
     our_base = os.environ['AUTOPKGTEST_BASE'] + '/lib'
@@ -58,9 +59,19 @@ qemu_cmd_default = None
 def parse_args():
     global args, qemu_cmd_default
 
-    parser = ArgumentParser()
+    mappings = { 'i[3456]86': 'i386' }
+    arch = os.uname()[4]
+    qemu_arch = None
+    for pattern, real_name in mappings.items():
+        if re.match('^' + pattern + '$', arch):
+            qemu_arch = real_name
+            break
+    if qemu_arch is None:
+        qemu_arch = arch
 
-    qemu_cmd_default = 'qemu-system-' + os.uname()[4]
+    qemu_cmd_default = 'qemu-system-' + qemu_arch
+
+    parser = ArgumentParser()
 
     parser.add_argument('-q', '--qemu-command', default=qemu_cmd_default,
                         help='QEMU command (default: %s)' % qemu_cmd_default)
-- 
2.5.0

